kong/tools/utils.lua

```diff
@@ -786,7 +786,7 @@ _M.format_host = function(p1, p2)
     return nil, "cannot format type '" .. t .. "'"
   end
   if typ == "ipv6" and not find(host, "[", nil, true) then
-    return "[" .. host .. "]" .. (port and ":" .. port or "")
+    return "[" .. _M.normalize_ipv6(host) .. "]" .. (port and ":" .. port or "")
```

Comment:
This is not ok.

There are 3 functions for normalizing/validating, and there is 1 for formatting (this one). With this change the formatter:

- normalizes/validates IPv6
- does not normalize/validate IPv4 nor names
- suddenly becomes a lot more expensive, due to the 1st point above

Reply:
All of our other "normalization" functions don't actually do any normalization. They just do validation.

Reply:
If we change behaviour, it should be consistent. Currently it moved the other direction

Reply:
Also the documentation is no longer correct add the behaviour of the function changed.